### ISSUE TITLE
Does not work on latest iojs in windows.

### ISSUE THREAD

**Issue by johnhaley81** [rvagg/pangyp#4]
If you compile the [hello-world example](https://github.com/joyent/node/tree/master/test/addons/hello-world) using pangyp in iojs 1.5.x on Windows it will compile successfully.

However if you then do `node test` you'll get the following console output:

```bash
C:\git-projects\node\test\addons\hello-world (master)
λ node test.js
module.js:339
  Module._extensions[extension](this, filename);
                               ^
Error: Module did not self-register.
    at Error (native)
    at Module.load (module.js:339:32)
    at Function.Module._load (module.js:294:12)
    at Module.require (module.js:349:17)
    at require (module.js:368:17)
    at Object.<anonymous> (C:\git-projects\node\test\addons\hello-world\test.js:2:15)
    at Module._compile (module.js:444:26)
    at Object.Module._extensions..js (module.js:462:10)
    at Module.load (module.js:339:32)
    at Function.Module._load (module.js:294:12)

C:\git-projects\node\test\addons\hello-world (master)
λ
```

**Comment by mgroshans** [comment]
I just spent an hour banging my head against the same problem. It literally works if you use `iojs test` instead of `node test`

No I am not joking, and yes they are the same binary... I have no idea how this works.

**Comment by am11** [comment]
That is a known issue with io.js, which is recently fixed. See https://github.com/iojs/io.js/issues/751.
I think we should port the patch https://github.com/iojs/io.js/pull/1251 and https://github.com/iojs/io.js/pull/1266 to pangyp.
Note, this optional linker hook is good enough for most of the repos, but it is not the ultimate fix: it would not work if your DLL export data  (virtual inheritance etc.).

**Comment by am11** [comment]
This is fixed in v2.2.0.
